### PR TITLE
Set leader metric after controller is functional

### PR DESCRIPTION
**Pull request by xpivarc**
**What this PR does / why we need it**:
Move exposing leader Prometheus metric after the controller is functional and tighten it up with a unit test.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
I think we can remove the `/leader` endpoint.

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/virt-controller/watch/application_test.go [hunk 1]
```diff
@@ -45,6 +64,108 @@ func newValidGetRequest() *http.Request {
...
+		pdbInformer, _ := testutils.NewFakeInformerFor(&v1beta1.PodDisruptionBudget{})
+		podInformer, _ := testutils.NewFakeInformerFor(&k8sv1.Pod{})
+		pvcInformer, _ := testutils.NewFakeInformerFor(&k8sv1.PersistentVolumeClaim{})
+		dataVolumeInformer, _ := testutils.NewFakeInformerFor(&cdiv1.DataVolume{})
+		rsInformer, _ := testutils.NewFakeInformerFor(&v1.VirtualMachineInstanceReplicaSet{})
+		storageClassInformer, _ := testutils.NewFakeInformerFor(&storagev1.StorageClass{})
+		crdInformer, _ := testutils.NewFakeInformerFor(&extv1beta1.CustomResourceDefinition{})
+		vmRestoreInformer, _ := testutils.NewFakeInformerFor(&snapshotv1.VirtualMachineRestore{})
+
+		var qemuGid int64 = 107
```
Comment (dhiller) [inline pkg/virt-controller/watch/application_test.go:91]: This isn't defined anywhere in production code?

**Review by dhiller — APPROVED** [review]
Looks good to me so far, one question though.

/lgtm

File: pkg/virt-controller/watch/application_test.go (hunk 1 above)
Comment (xpivarc) [reply pkg/virt-controller/watch/application_test.go:91]: Do you mean the `qemuGid`? We have a default in `application.go` called `defaultLauncherSubGid`. The test doesn't rely on this.

**Pull request merged**